Cargo.toml

```diff
@@ -11,4 +11,4 @@ panic = "abort"
 lto = true
 
 [patch.crates-io]
-kvm-bindings = { git = "https://github.com/firecracker-microvm/kvm-bindings", tag = "v0.6.0-1", features = ["fam-wrappers"] }
+kvm-bindings = { git = "https://github.com/firecracker-microvm/kvm-bindings", tag = "v0.6.0-2", features = ["fam-wrappers"] }
```

Comment:
I am not sure what do you mean by `refactoring` here. The addition of `vcpu_features` requires all the changes to be in one place, because change in `vcpu.init` needs this. I guess I can make an update of `kvm-bindings` and `kvm-ioctls` to be in separate commit.

Reply:
What I mean is even though the addition of `vcpu_features` requires all the refactoring, the actual functional change (adding `vcpu_features` argument to the `vcpu::init` and adding features to `kvi`) can be done in a separate commit. Thus the reviewer can verify that the 1st commit just preserves the original behaviour and the 2nd commit adds the functionality it claims. If both changes are in the same commit, the reviewer needs to do both operations simultaneously, which is less efficient (either they spend more time or sacrifice the quality of the review).

Reply:
So the first commit will add `vcpu_features` parameter to `vcpu::init` method that is not used anywhere and reviewer will not know why this is needed until they look at the next commit, which actually adds usage to this parameter. I don't think this makes sense. Keeping all related to each other changes in one commit gives the reviewer to observe all changes in one go.

Reply:
When needed, you can redirect reviewers to the next commit by saying like "see the next commit for more details" in the commit message (of course, the commit message should provide "why" in as self-contained a way as possible). I saw patch series following this style multiple times in kernel development and you can see [here](https://lore.kernel.org/all/?q=%22No%20functional%20changes%22) (just by querying with "No functional changes"). When reviewers want to see all changes in one go, they can take a look in PR view.

Reply:
Split this commit into several

Reply:
Thanks for splitting, appreciate it.
`Vcpu restoration is broken in this commit.` -- does it affect only integration tests or also unit tests? We aim to keep them passing for every commit.